modules/sdk/project-templates-war-hook/src/main/resources/archetype-resources/src/main/java/__className__.java

```diff
@@ -0,0 +1,17 @@
+package ${package}.${className};
+
+import com.liferay.portal.kernel.events.SimpleAction;
+import com.liferay.portal.kernel.events.ActionException;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+/**
+ * @author ${author}
+ */
+public class ${className} extends SimpleAction {
+	
+	@Override
+	public void run(String[] arg0) throws ActionException {
```

Comment:
Lets add two different type of actions

looking here: https://dev.liferay.com/develop/tutorials/-/knowledge_base/6-2/performing-a-custom-action-using-a-hook

Lets create a global.startup.events=${package}.${className}GlobalStartupAction  simple action that runs when portal boots up

And then create a login.pre.action like this
```
public class LoginAction extends Action {
    public void run(HttpServletRequest req, HttpServletResponse res) {
        System.out.println("## My custom login action");
    }
}
```

So portal.properties will have
login.pre.action=xxxx
global.startup.events=xxxx

just to show the two common types of hook actions.